lib/impl/submissionExportHelper/csvHeaders.js

```diff
@@ -109,18 +111,40 @@ function generateCSVHeaders(fieldKeys, mergedFieldEntries, fieldHeader) {
     csv += headerName + ',';
   });
 
-  //for each form get each of the unique fields and add a header
+  var fieldKeysProcessed = [];
+  var fieldsProcessed = {};
   fieldKeys.forEach(function(fieldKey) {
-    // check if its a repeating form (and extra headers required)
+  // check if its a repeating form (and extra headers required)
     var field = mergedFieldEntries[fieldKey];
+    if (field.type === 'sectionBreak' && field.repeating) {
+      var fieldsInThisSection = sectionUtils.getFieldsInSection(field._id, _.values(mergedFieldEntries));
+      _.each(fieldsInThisSection, function(field) {
+        fieldsProcessed[field._id] = true;
+      });
+
+      for (var i = 0; i < field.fieldOptions.definition.maxRepeat; i++) {
+        _.each(fieldsInThisSection, function(fieldInThisSection) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Define this function outside of a loop. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionDefinitionInsideLoop)